saphana-plugin/widgets/SapHana-batchsource.json

```diff
@@ -0,0 +1,164 @@
+{
+  "metadata": {
+    "spec-version": "1.0"
+  },
+  "display-name": "SAP HANA",
+  "configuration-groups": [
+    {
+      "label": "Basic",
+      "properties": [
+        {
+          "widget-type": "textbox",
+          "label": "Reference Name",
+          "name": "referenceName",
+          "widget-attributes": {
+            "placeholder": "Name used to identify this source for lineage"
+          }
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Driver Name",
+          "name": "jdbcPluginName",
+          "widget-attributes": {
+            "default": "sap"
+          }
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Host",
+          "name": "host",
+          "widget-attributes": {
+            "default": "localhost"
+          }
+        },
+        {
+          "widget-type": "number",
+          "label": "Port",
+          "name": "port",
+          "widget-attributes": {
+            "default": "39017"
+          }
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Database",
+          "name": "database"
+        },
+        {
+          "widget-type": "textarea",
+          "label": "Import Query",
+          "name": "importQuery",
+          "widget-attributes": {
+            "rows": "4"
+          },
+          "plugin-function": {
+            "widget": "outputSchema",
+            "output-property": "schema",
+            "omit-properties": [
+              {
+                "name": "schema"
+              }
+            ],
+            "add-properties": [
+              {
+                "name": "query",
+                "plugin-property-for-value": "importQuery",
+                "widget-type": "textarea",
+                "label": "Query",
+                "widget-attributes": {
+                  "rows": "4"
+                }
+              }
+            ]
+          }
+        },
+        {
+          "widget-type": "textarea",
+          "label": "Bounding Query",
+          "name": "boundingQuery",
+          "widget-attributes": {
+            "rows": "4"
+          }
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Split-By Field Name",
+          "name": "splitBy"
```

Comment:
If this value has to be one of the fields from the input schema then I would recommend use `input-field-selector` widget. This pre-populates fields from input schema in a dropdown.

Reply:
done